distributed_shampoo/tests/distributed_shampoo_test.py

```diff
@@ -55,6 +55,21 @@ def test_invalid_preconditioner_config(self) -> None:
                 preconditioner_config=DefaultShampooConfig,
             )
 
+        with mock.patch.object(
+            distributed_shampoo,
+            "isinstance",
+            return_value=False,
+        ):
```

Comment:
What is the reason for using `mock` here instead of just passing an invalid config option, e.g. `DefaultEigenConfig`?

Reply:
Oh sorry, I meant something like `DefaultShampooConfig` (or any other config instance that is not inheriting from `MatrixFunctionConfig`).

Reply:
Makes sense, but I think it would be justified to add an ignore statement to this line here, since it allows us to directly test the desired behavior in a way that is robust to future changes. E.g., imagine there will be an `isinstance` call before the one we want to test in the flow of the program in the future, then the test would fail (error message doesn't match anymore) and has to be fixed manually.

But I also don't really mind using `mock`, was just curious about the reasoning.